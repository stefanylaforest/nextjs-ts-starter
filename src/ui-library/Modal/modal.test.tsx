import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Modal from './modal';

const onClick = jest.fn();

describe('Modal', (): void => {
  it('should render without crashing ', () => {
    render(
      <Modal show={true} closeModal={onClick}>
        <p>A message</p>
      </Modal>
    );
    expect(screen.getByTestId('modal')).toMatchSnapshot();
  });

  it('should show if "show" prop is true', () => {
    render(
      <Modal show={true} closeModal={onClick}>
        <p>A message</p>
      </Modal>
    );
    expect(screen.getByTestId('modal')).toBeInTheDocument();
  });

  it('show not render if prop is false', () => {
    const { container } = render(
      <Modal show={false} closeModal={onClick}>
        <p>Modal Message</p>
      </Modal>
    );
    expect(container).toBeEmptyDOMElement();
  });

  it('should render a close button', () => {
    render(
      <Modal show={true} closeModal={onClick}>
        <p>A message</p>
      </Modal>
    );
    expect(screen.getByRole('button', { name: 'Close Modal' })).toBeInTheDocument();
  });

  it('should render a title', async () => {
    render(
      <Modal show={true} closeModal={onClick} title={'A title'}>
        <p>A message</p>
      </Modal>
    );
    expect(await screen.findByRole('heading', { name: 'A title' })).toBeInTheDocument();
  });

  it('should render a footer', async () => {
    render(
      <Modal
        show={true}
        closeModal={onClick}
        footer={
          <div>
            <button>ok</button>
            <button>cancel</button>
          </div>
        }
      >
        <p>A message</p>
      </Modal>
    );
    expect(screen.getByTestId('modal-actions')).toBeInTheDocument();
  });

  it('should close modal on click of close button', async () => {
    render(
      <Modal show={true} closeModal={onClick}>
        <p>A message</p>
      </Modal>
    );
    const closeBtn = screen.getByRole('button', { name: 'Close Modal' });
    await userEvent.click(closeBtn);
    expect(onClick).toHaveBeenCalled();
  });

  it('should close close modal on outside click', async () => {
    const onClose = jest.fn();
    render(
      <Modal show={true} closeModal={onClose}>
        <p>A message</p>
      </Modal>
    );
    const modalContainer = screen.getByTestId('modal');
    await userEvent.click(modalContainer);
    expect(onClose).toHaveBeenCalled();
  });
});
