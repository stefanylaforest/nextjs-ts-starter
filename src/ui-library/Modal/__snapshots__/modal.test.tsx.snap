// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render without crashing  1`] = `
<div
  class="bodyOverlay"
  data-testid="modal"
  role="button"
>
  <div
    class="modalWrapper"
    role="dialog"
  >
    <div
      class="header"
    >
      <button
        aria-label="Close Modal"
        class="icon"
        type="button"
      >
        <svg
          class="h-6 w-6"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
    <div
      class="body"
    >
      <p>
        A message
      </p>
    </div>
  </div>
</div>
`;
